weblate/checks/consistency.py

```diff
@@ -95,9 +95,14 @@ class TranslatedCheck(TargetCheck):
 
     check_id = "translated"
     name = _("Has been translated")
-    description = _("This string has been translated in the past")
+    description = ""
     ignore_untranslated = False
 
+    def get_description(self, check=None):
+        unit = check.unit
+        self.check_target_unit(unit.source, unit.target, unit)
+        return self.description
```

Comment:
Beware that the in database check state might be outdated, so the check_target_unit might return False when get_description is called. In this case you return random description of previously triggered check.

Reply:
What is the meaning of "check state might be outdated"

Reply:
The get_description is called on the translate page when the check object is stored in the database. Meanwhile the string can be updated and check update in progress in the background. So the get_description should work without problems when the check is no longer failing.